### PR TITLE
Generate documentation (swagger2markup) and Swagger annotation tuning

### PR DESCRIPTION
**Pull request by eputtone**

### CODE REVIEW

File: nflow-tests/src/test/java/com/nitorcreations/nflow/tests/Swagger2MarkupTest.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  public Swagger2MarkupTest() {
+    super(server);
+  }
+
+  @Test
+  public void convertRemoteSwaggerToAsciiDoc() throws IOException {
+    Swagger2MarkupConverter.from(server.getHttpAddress() + "/api/swagger.json").build().intoFolder(SWAGGER2_MARKUP_ASCIIDOC_DIR);
+
+    // Then validate that three AsciiDoc files have been created
```
Comment (efonsell) [inline nflow-tests/src/test/java/com/nitorcreations/nflow/tests/Swagger2MarkupTest.java]: Three or four?

File: pom.xml [hunk 2]
```diff
@@ -684,9 +685,15 @@
         <version>${swagger.version}</version>
       </dependency>
       <dependency>
-        <groupId>io.swagger</groupId>
```
Comment (efonsell) [inline pom.xml]: why is this removed?

File: nflow-tests/src/test/java/com/nitorcreations/nflow/tests/Swagger2MarkupTest.java (hunk 1 above)
Comment (eputtone) [inline nflow-tests/src/test/java/com/nitorcreations/nflow/tests/Swagger2MarkupTest.java]: three generated files + index.adoc = total of 4 files

File: pom.xml (hunk 2 above)
Comment (eputtone) [inline pom.xml]: it is a duplicate dependency - see 10 lines above
